packages/core/src/client/index.ts

```diff
@@ -204,7 +207,8 @@ export class Web3InboxClient {
     projectId: string;
     domain?: string;
     allApps?: boolean;
-    logLevel?: "error" | "info" | "debug"
+    logLevel?: "error" | "info" | "debug";
+    rpcUrl?: string;
```

Comment:
This doesn't seem like right API. This takes any rpcUrl, but it expects it to be a Blockchain API URL and will format it as such.

Instead, this should be a callback that given a chain ID, it returns a URL. By default, that's a callback that generates a URL containing the project ID.

Reply:
So users of the SDK should supply a callback like:

```
(chainId: string) => `https://their-favorite-rpc/${chainId}`
```

And by default it's
```
(chainId: string) => `https://rpc.walletconnect.com?projectId=${this.projectId}&chainId=${chainId}`
```

That's doable